src/Http/Http.Extensions/gen/RequestDelegateGeneratorSources.cs

```diff
@@ -549,8 +559,29 @@ namespace Microsoft.AspNetCore.Http.Generated
 
     file static class GeneratedRouteBuilderExtensionsCore
     {
-{{GetGenericThunks(genericThunks)}}
-{{GetThunks(thunks)}}
+        private static readonly string[] GetVerb = new[] { global::Microsoft.AspNetCore.Http.HttpMethods.Get };
+        private static readonly string[] PostVerb = new[] { global::Microsoft.AspNetCore.Http.HttpMethods.Post };
+        private static readonly string[] PutVerb = new[]  { global::Microsoft.AspNetCore.Http.HttpMethods.Put };
+        private static readonly string[] DeleteVerb = new[] { global::Microsoft.AspNetCore.Http.HttpMethods.Delete };
+        private static readonly string[] PatchVerb = new[] { global::Microsoft.AspNetCore.Http.HttpMethods.Patch };
+
+        {{endpoints}}
+
+        internal static RouteHandlerBuilder MapCore(
+            this IEndpointRouteBuilder routes,
+            string pattern,
+            Delegate handler,
+            IEnumerable<string>? httpMethods,
+            MetadataPopulator populateMetadata,
+            RequestDelegateFactoryFunc createRequestDelegate)
+        {
+            return RouteHandlerServices.Map(routes, pattern, handler, httpMethods, populateMetadata, createRequestDelegate);
+        }
+
+        private static T Cast<T>(Delegate d, T _) where T : Delegate
```

Comment:
Why?

Reply:
See #5 in https://github.com/dotnet/aspnetcore/issues/48289. It's a reaction to https://github.com/dotnet/csharplang/blob/main/proposals/lambda-method-group-defaults.md#breaking-change

Reply:
OK, why do you need a method at all? Everything is local variables and everything is captured in a closure right? e.g

Instead of:

```C#
var handler = Cast(del, global::System.String (global::Microsoft.AspNetCore.Http.HttpRequest arg0, global::Microsoft.AspNetCore.Http.HttpResponse arg1) => throw null!);
```

You could do:

```C#
var handler = global::System.String (global::Microsoft.AspNetCore.Http.HttpRequest arg0, global::Microsoft.AspNetCore.Http.HttpResponse arg1) => throw null!;
```
Since we're not generating overloads anymore, why do we need to cast?

Reply:
I'm still confused. How does a direct cast work even in this case?

Reply:
This is the same problem as anonymous types now.

Reply:
I see how this works now, very nice but also crazy, I approve.